var/spack/repos/builtin/packages/py-installer/package.py

```diff
@@ -3,12 +3,10 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-import os
-
 from spack.package import *
 
 
-class PyInstaller(Package, PythonExtension):
+class PyInstaller(PythonPackage):
```

Comment:
This was done on purpose. `py-installer` is a possible replacement for `py-pip` that is 1% of the size and complexity and literally only copies a wheel to an installation directory, nothing else. One of my hopes was to add `python_installer` in addition to `python_pip` as a possible build backend. This would be much simpler than pip since it wouldn't require an entire package manager with a ton of flags to disable all of its features. `py-build` is the other half of the equation, and can be used to actually build wheels (again, without an entire package manager, e.g., pip). However, I never did this because build has dozens of dependencies. Previously, these would need to be built (e.g., by pip), but now that we're installing wheels, this has once again become a reality.

Reply:
So the suggestion is to drop py-installer from this pr?

Reply:
Yes. It should be installable without pip.